portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,13 +37,28 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String slimit = request.getParameter("limit");
+    int limit;
+
+    try {
+      limit = Integer.parseInt(slimit);
+      if(limit<0)
+        limit = Integer.MAX_VALUE;
+    } catch (Exception e) {
+      limit = Integer.MAX_VALUE;
+    }
+
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
-
+    int ct=0;         //counter for limiting number of results. 
     List<Comment> comments = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
+      if(ct >= limit)
```

Comment:
nit: In Google we always use {} for if statements, even if there is only one line in them..